metadata/metadata.js

```diff
@@ -7,70 +7,113 @@
  */
 
 const osgi = require('../osgi');
-const utils = require('../utils');
 const log = require('../log')('metadata');
 
 let MetadataRegistry = osgi.getService("org.openhab.core.items.MetadataRegistry");
 let Metadata = Java.type("org.openhab.core.items.Metadata");
 let MetadataKey = Java.type("org.openhab.core.items.MetadataKey");
 
-
 /**
- * This function will return the Metadata object associated with the
- * specified Item.
+ * Class representing an Item's Metadata
  * 
- * @memberof metadata
- * @param {String} name of the Item
- * @param {String} namespace name of the namespace
- * @returns {String|null} the metadata as a string, or null
+ * @memberOf metadata
  */
-let getValue = function(itemName, namespace) {
-    let result = MetadataRegistry.get(new MetadataKey(namespace, itemName));
-    return result ? result.value : null;
-};
+class ItemMetadata {
 
-let addValue = function(itemName, namespace, value) {
-    let key = new MetadataKey(namespace, itemName);
-    MetadataRegistry.add(new Metadata(key, value, {}));
-}
+    /**
+     * Don't use this constructor, instead call {@link items.Item.metadata}.
+     * @param {String} itemName
+     * @hideconstructor
+     */
+    constructor(itemName) {
+        this.itemName = itemName;
+    }
 
-let updateValue = function(itemName, namespace, value) {
-    let metadata = createMetadata(itemName, namespace, value);
-    let result = MetadataRegistry.update(metadata);
-    return result ? result.value : null;
-}
+    /**
+     * This function will return the Metadata value associated with the
+     * specified Item.
+     * 
+     * @param {String} namespace name of the namespace
+     * @returns {String|null} the metadata value as a string, or null
+     */
+    getValue(namespace) {
+        let result = MetadataRegistry.get(new MetadataKey(namespace, this.itemName));
+        return result ? result.value : null;
+    };
 
-/**
- * Adds (inserts) or updates a metadata value.
- * 
- * @param {String} itemName the name of the item
- * @param {String} namespace the name of the namespace
- * @param {String} value the value to insert or update
- * @returns {Boolean} true if the value was added, false if it was updated
- */
-let upsertValue = function(itemName, namespace, value) {
-    let existing = getValue(itemName, namespace);
+    /**
+    * Adds (inserts) a metadata value.
+    * 
+    * @param {String} namespace the name of the namespace
+    * @param {String} value the value to insert or update
+    */
+    addValue(namespace, value) {
```

Comment:
I think it's worth to take a full potential of metadata and let the `addValue` method to set also metadata configuration values. currently this method sets it to empty. I was thinking about splitting it into two methods as for the reading configuration values, but apparently the `Metadata` object doesn't have any methods - it's possible only via constructor